lib/extras/dec/jxl.cc

```diff
@@ -516,6 +516,14 @@ bool DecodeImageJXL(const uint8_t* bytes, size_t bytes_size,
     }
   }
   boxes.FinalizeOutput();
+  // Encoded Exif has a 4-byte offset, but for processing we expect the TIFF
+  // header at the start of the data.
+  constexpr uint8_t zero32[] = {0, 0, 0, 0};
+  if (ppf->metadata.exif.size() >= 4 &&
+      memcmp(ppf->metadata.exif.data(), zero32, 4) == 0) {
+    ppf->metadata.exif.erase(ppf->metadata.exif.begin(),
+                             ppf->metadata.exif.begin() + 4);
+  }
```

Comment:
This probably needs an `else` to return error in case Exif does not start with the expected 0 offset.

Reply:
Is it OK for that to cause djxl to fail? `JxlEncoderAddBox` makes it very easy to create an invalid Exif box. I could add a check in there too, so that it insists on leading zeros when the type is Exif, but it could still be an issue for existing files that were previously created through the API (although of course they would still be decodable through the API).

Reply:
Well I think there are two options:

- either it always returns the actual Exif starting with the tiff header — so stripping the zero offset, or skipping [offset] amount of bytes after the offset field if the offset is not zero
- or it simply returns the Exif box as it is, including the offset and anything that goes before the tiff header (but then anything that encodes Exif in png or jpeg probably has to strip the pre-tiff-header stuff because in those formats there cannot be such a thing)

I think djxl should not fail if the offset is nonzero, but it probably should then skip the bytes before the tiff header so the output format can store Exif properly also in that case.

Perhaps it would be good to skip to where the tiff header is supposed to be, check that there is indeed a tiff header (49 49 2A 00 or 4D 4D 00 2A) at that position, and issue a warning that there's an invalid Exif box otherwise. I suppose djxl can still continue in this case, and either just write a bad exif blob or write no exif at all.

Reply:
Yes, it's specified to be a big-endian uint32.

Reply:
LGTM. Thanks!